tensorboard/webapp/app_routing/effects/app_routing_effects_test.ts

```diff
@@ -651,7 +651,42 @@ describe('app_routing_effects', () => {
       expect(replaceStateSpy).toHaveBeenCalledWith('/experiments#foo');
     });
 
-    it('does not preserve hash upon replace for non-initial navigation', () => {
+    it('preserves hash upon navigations to the same route id', () => {
```

Comment:
in this behavioral spec, maybe specify that this is a temporary behavior until all apps use the router?

Reply:
Done.